src/ocpp.cpp

```diff
@@ -42,99 +44,125 @@ void ArduinoOcppTask::begin(EvseManager &evse, LcdTask &lcd, EventLog &eventLog,
-        if (response["status"].as<String>().equals("Accepted")) {
-            LCD_DISPLAY("OCPP connected!");
-            bootNotificationAccepted = true;
-        } else {
-            LCD_DISPLAY("OCPP refused EVSE");
-        }
+    ArduinoOcpp::configuration_init(filesystem);
+    ArduinoOcpp::declareConfiguration<const char*>(
+        "MeterValuesSampledData", "Power.Active.Import,Energy.Active.Import.Register,Current.Import,Current.Offered,Voltage,Temperature");
+    ArduinoOcpp::declareConfiguration<bool>(
+        "AO_PreBootTransactions", true);
+    
+    /*
+     * Initialize the OCPP library and provide it with the charger credentials
+     */
+    ocpp_initialize(*ocppSocket, ChargerCredentials(
+            "Advanced Series",         //chargePointModel
+            "OpenEVSE",                //chargePointVendor
+            currentfirmware.c_str(),   //firmwareVersion
+            serial.c_str(),            //chargePointSerialNumber
+            evse->getFirmwareVersion() //meterSerialNumber
+        ), filesystem);
+
+    /*
+     * Load OCPP configs. Default values will be overwritten by OpenEVSE configs. Mark configs
+     * to require reboot if changed via OCPP server
+     */
+    backendUrl = ArduinoOcpp::declareConfiguration<const char*>("AO_BackendUrl", "", AO_FILENAME_PREFIX "ocpp-creds.jsn", true, true, true, true);
+    chargeBoxId = ArduinoOcpp::declareConfiguration<const char*>("AO_ChargeBoxId", "", AO_FILENAME_PREFIX "ocpp-creds.jsn", true, true, true, true);
+    authKey = ArduinoOcpp::declareConfiguration<const char*>("AuthorizationKey", "", AO_FILENAME_PREFIX "ocpp-creds.jsn", true, true, true, true);
+    freevendActive = ArduinoOcpp::declareConfiguration<bool>("AO_FreeVendActive", true, CONFIGURATION_FN, true, true, true, true);
+    freevendIdTag = ArduinoOcpp::declareConfiguration<const char*>("AO_FreeVendIdTag", "DefaultIdTag", CONFIGURATION_FN, true, true, true, true);
+    allowOfflineTxForUnknownId = ArduinoOcpp::declareConfiguration<bool>("AllowOfflineTxForUnknownId", true, CONFIGURATION_FN, true, true, true, true);
+    silentOfflineTx = ArduinoOcpp::declareConfiguration<bool>("AO_SilentOfflineTransactions", true, CONFIGURATION_FN, true, true, true, true);
+
+    //when the OCPP server updates the configs, the following callback will apply them to the OpenEVSE configs
+    setOnReceiveRequest("ChangeConfiguration", [this] (JsonObject) {
+        DynamicJsonDocument updateQuery (JSON_OBJECT_SIZE(6));
+        updateQuery["ocpp_server"] = (const char*) *backendUrl;
+        updateQuery["ocpp_chargeBoxId"] = (const char*) *chargeBoxId;
+        updateQuery["ocpp_authkey"] = (const char*) *authKey;
+        updateQuery["ocpp_auth_auto"] = *freevendActive ? 1 : 0;
+        updateQuery["ocpp_idtag"] = (const char*) *freevendIdTag;
+        updateQuery["ocpp_auth_offline"] = *allowOfflineTxForUnknownId ? 1 : 0;
+        config_deserialize(updateQuery);
+        config_commit();
```

Comment:
Would be better to use `config_set` rather than using JSON

Reply:
Done